packages/flutter/lib/src/material/text_button.dart

```diff
@@ -377,18 +379,30 @@ class _TextButtonWithIconChild extends StatelessWidget {
     Key? key,
     required this.label,
     required this.icon,
+    required this.iconAlignment,
   }) : super(key: key);
 
   final Widget label;
   final Widget icon;
+  final ButtonIconAlignment iconAlignment;
 
   @override
   Widget build(BuildContext context) {
     final double scale = MediaQuery.maybeOf(context)?.textScaleFactor ?? 1;
     final double gap = scale <= 1 ? 8 : lerpDouble(8, 4, math.min(scale - 1, 1))!;
+
+    List<Widget> _renderButtonChildren() {
+      switch (iconAlignment) {
+        case ButtonIconAlignment.start:
+          return <Widget>[icon, SizedBox(width: gap), label];
+        case ButtonIconAlignment.end:
+          return <Widget>[label, SizedBox(width: gap), icon];
+      }
+    }
+
```

Comment:
drive-by nit: this would probably be more performant if you set a variable rather than using a function here.

Reply:
Sure, thanks. Will do the change 🚀

Reply:
I am trying to find a more performant way but can't, can you reply with an example if that is possible 😊🙌